packages/google-pay-integration/src/google-pay-paypal-commerce/google-pay-paypal-commerce-gateway.ts

```diff
@@ -0,0 +1,107 @@
+import {
+    InvalidArgumentError,
+    PaymentIntegrationService,
+    PaymentMethod,
+} from '@bigcommerce/checkout-sdk/payment-integration-api';
+
+import GooglePayGateway from '../gateways/google-pay-gateway';
+import {
+    GooglePayCardDataResponse,
+    GooglePayPayPalCommerceGatewayParameters,
+    GooglePayPayPalCommerceInitializationData,
+    GooglePaySetExternalCheckoutData,
+} from '../types';
+
+import PayPalCommerceScriptLoader from './google-pay-paypal-commerce-script-loader';
+import { TGooglePayConfig, TPayPalGoogleSdk } from './types';
+
+export default class GooglePayPaypalCommerceGateway extends GooglePayGateway {
+    private paypalGoogleSdk?: TPayPalGoogleSdk;
+    private googlepayConfig?: TGooglePayConfig;
+    private paymentIntegrationService: PaymentIntegrationService;
+    private paymentMethod?: PaymentMethod<GooglePayPayPalCommerceInitializationData>;
+
+    constructor(
+        service: PaymentIntegrationService,
+        private paypalCommerceScriptLoader: PayPalCommerceScriptLoader,
+    ) {
+        super('paypalsb', service);
+        this.paymentIntegrationService = service;
+    }
+
+    async initialize(
+        getPaymentMethod: () => PaymentMethod<GooglePayPayPalCommerceInitializationData>,
+    ): Promise<void> {
+        await super.initialize(getPaymentMethod);
+
+        const paymentMethod = await this.paymentIntegrationService.loadPaymentMethod(
+            'googlepaypaypalcommerce',
+        );
+
+        if (!paymentMethod) {
+            throw new InvalidArgumentError('Unable to initialize payment method.');
```

Comment:
Usually in this case we use `throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod)`